tests/test_add_model.py

```diff
@@ -0,0 +1,423 @@
+    mock_isinstance.assert_called_with(model_class, swagger._Nested)
+    assert mock_deduce_swagger_type.call_count == len(
+        model_class.resource_fields.items()
+    )
+
+
+# Setup test for:
+#     * resource_fields: YES
+#     * nested subclass: YES
+#     * __init__: NO
+#     * swagger_metadata:YES
+
+test_fixtures_model_inputs = [
+    ModelWithResourceFieldsWithRequiredWithSwaggerMetadata,
+]
+
+
+@patch("flask_restful_swagger.swagger.registry")
+@patch("flask_restful_swagger.swagger.deduce_swagger_type")
+@patch("flask_restful_swagger.swagger.isinstance")
+@patch("flask_restful_swagger.swagger.hasattr")
+@patch("flask_restful_swagger.swagger.dir")
+@patch("flask_restful_swagger.swagger._parse_doc")
+@pytest.mark.parametrize("model_class", test_fixtures_model_inputs)
+def test_add_model_with_resource_fields_nested_swagger_metadata(
+    mock_parse_doc,
+    mock_dir,
+    mock_hasattr,
+    mock_isinstance,
+    mock_deduce_swagger_type,
+    mock_registry,
+    model_class,
+):
+    _mock_registry = {"models": {}}
+    mock_registry.__getitem__.side_effect = _mock_registry.__getitem__
+    mock_registry.__setitem__.side_effect = _mock_registry.__setitem__
+    mock_parse_doc.return_value = (None, None)
+    mock_hasattr.return_value = True
+    mock_dir.return_value = ["resource_fields"]
+    mock_isinstance.return_value = True
+    mock_deduce_swagger_type.return_value = "dummy_swagger_type"
+
+    swagger.add_model(model_class)
+
+    mock_dir.assert_called_with(model_class)
+    assert mock_dir.call_count == 2
+    mock_hasattr.assert_called_once_with(model_class, "required")
+    mock_isinstance.assert_called_with(model_class, swagger._Nested)
+    assert mock_deduce_swagger_type.call_count == len(
+        model_class.resource_fields.items()
+    )
+
+
+# Setup test for:
+#     * resource_fields: NO
+#     * nested subclass: NO
+#     * __init__: YES
+#     * swagger_metadata: NO
+
+test_fixtures_model_inputs = [
+    MockBasicObject,
+    TodoItem,
+]
+
+
+@patch("flask_restful_swagger.swagger.registry")
+@patch("flask_restful_swagger.swagger.inspect.getargspec")
+@patch("flask_restful_swagger.swagger.dir")
+@patch("flask_restful_swagger.swagger._parse_doc")
+@pytest.mark.parametrize("model_class", test_fixtures_model_inputs)
+def test_add_model_init(
+    mock_parse_doc, mock_dir, mock_getargspec, mock_registry, model_class
+):
+    _mock_registry = {"models": {}}
+    mock_registry.__getitem__.side_effect = _mock_registry.__getitem__
+    mock_registry.__setitem__.side_effect = _mock_registry.__setitem__
+    mock_parse_doc.return_value = (None, None)
+    mock_dir.return_value = ["__init__"]
+
+    mock_argspec = Mock()
+    mock_argspec.args = ["self", "arg1", "arg2", "arg3"]
+    mock_argspec.defaults = ("123",)
+    mock_getargspec.return_value = mock_argspec
+
+    swagger.add_model(model_class)
+
+    mock_getargspec.assert_called_once_with(model_class.__init__)
```

Comment:
This only tests the happy path.  Should we assert that material inside the 'resource_fields' path is not called?   I think we could add some of the mocks from the pervious test and accomplish this.  (Even use the same setup function...)

Reply:
Good catch.  I added a few more mocking statements and checking they are not called:
```
def test_add_model_init(model_class):
        #...
        mock_hasattr.assert_not_called()
        mock_deduce_swagger_type.assert_not_called()
```